### PR TITLE
Feat/backend ci workflow

### PR DESCRIPTION
**Pull request by SpicyI**
this PR: 
 - add check backend job to ci workflow

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -26,10 +27,30 @@ jobs:
...
-
+  build_and_test_backend:
+    docker:
+      - image: cimg/python:3.12.4
+    steps:
+      - checkout
+      - run:
+          command: cd backend && poetry config virtualenvs.create false && poetry install
+          name: Install dependencies
+      - run:
```
Comment (Hattaki-Hajar) [inline .circleci/config.yml:38]: remove this

File: .circleci/config.yml [hunk 2]
```diff
@@ -26,10 +27,30 @@ jobs:
       - run:
           command: cd frontend && npm run lint
           name: Lint
-
+  build_and_test_backend:
+    docker:
+      - image: cimg/python:3.12.4
```
Comment (hmoumani) [inline .circleci/config.yml:32]: use same image as the one in backend

File: .circleci/config.yml [hunk 3]
```diff
@@ -26,10 +27,30 @@ jobs:
...
+    steps:
+      - checkout
+      - run:
+          command: cd backend && poetry config virtualenvs.create false && poetry install
```
Comment (hmoumani) [inline .circleci/config.yml:36]: ugly re usage of `cd backend` 😫

**Review by Hattaki-Hajar — CHANGES_REQUESTED** [review]
good job

**Review by hmoumani — CHANGES_REQUESTED** [review]
GG

File: .circleci/config.yml (hunk 2 above)
Comment (SpicyI) [reply .circleci/config.yml:32]: tm7wa

File: .circleci/config.yml (hunk 1 above)
Comment (SpicyI) [reply .circleci/config.yml:38]: good catch

File: .circleci/config.yml (hunk 3 above)
Comment (SpicyI) [reply .circleci/config.yml]: yeah i agree

**Review by Hattaki-Hajar — APPROVED** [review]

**Pull request merged**